File: codechef/CodeVaitam/Riddle.java
import java.util.Scanner;
import java.math.BigInteger;

public class Riddle
{
	private static char[] operands = new char[] {'/', '*', '+', '-'};

	public static void main(String[] args)
	{
		Scanner sc = new Scanner(System.in);

		int t = sc.nextInt();
		for (int i = 0; i < t; i++)
		{
			int n = sc.nextInt();

			BigInteger res = BigInteger.valueOf(sc.nextInt());
			BigInteger num;

			int cnt = 0, index = 0;
			for (int j = 1; j < n; j++) {
				char cur_op = operands[index];
				cnt++;
				if (cnt == 4) {
					cnt = 0;
					index = (index + 2) % 4;
				} else {
					index = (index + 1) % 4;
				}

				num = BigInteger.valueOf(sc.nextInt());

				if (cur_op == '+') 
					res = res.add(num);
				else if (cur_op == '-')
					res = res.subtract(num);
				else if (cur_op == '*')
					res = res.multiply(num);
				else 
					res = res.divide(num);
			}

			System.out.println(res.toString());
		}

	}
}

File: codechef/CodeVaitam/riddle.cpp
#include <iostream>

using namespace std;

typedef signed long long int64;

int main()
{
	ios_base::sync_with_stdio(false);

	char ops[4] = {'/', '*', '+', '-'};

	int t, n;
	cin >> t;
	for (int tt = 0; tt < t; tt++) {
		cin >> n;

		int64 res, num;
		cin >> res;

		int count = 0, index = 0;
		for (int i = 1; i < n; i++) {

			char cur_op = ops[index];
			count++;
			if (count == 4) {
				count = 0;
				index = (index + 2) % 4;
			} else {
				index = (index + 1) % 4;
			}

			cin >> num;
			
			if (cur_op == '+')
				res += num;
			else if (cur_op == '-')
				res -= num;
			else if (cur_op == '*')
				res *= num;
			else
				res /= num;
		}

		cout << res << endl;
	}
}